src/envoy/http/authn_wasm/config.cc

```diff
@@ -0,0 +1,113 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/envoy/http/authn_wasm/config.h"
+
+#include "authentication/v1alpha1/policy.pb.h"
+#include "google/protobuf/text_format.h"
+#include "google/protobuf/util/json_util.h"
+
+namespace iaapi = istio::authentication::v1alpha1;
+
+namespace Envoy {
+namespace Extensions {
+namespace Wasm {
+namespace AuthnWasm {
+
+// imports from the low-level API
+using Common::Wasm::Null::NullVmPluginFactory;
+using Common::Wasm::Null::Plugin::logDebug;
+using Common::Wasm::Null::Plugin::logError;
+
+using ::google::protobuf::TextFormat;
+using namespace google::protobuf::util;
+
+void PluginRootContext::onConfigure(std::unique_ptr<WasmData> configuration) {
+  logDebug("called PluginRootContext::onConfigure()");
+  JsonParseOptions json_options;
+  Status status = JsonStringToMessage(configuration->toString(),
+                                      &filter_config_, json_options);
+  if (status != Status::OK) {
+    logError("Cannot parse authentication filter config: " +
+             configuration->toString());
+  } else {
+    std::string out_str;
+    TextFormat::PrintToString(filter_config_, &out_str);
+    logDebug("Applied authentication filter config:\n" + out_str);
+  }
+}
+
+void PluginRootContext::onStart() {
+  logDebug("called PluginRootContext::onStart()");
+};
+
+void PluginRootContext::onTick() {
+  logDebug("called PluginRootContext::onTick()");
+};
+
+void PluginContext::onCreate() { logDebug("called PluginContext::onCreate()"); }
+
+Http::FilterHeadersStatus PluginContext::onRequestHeaders() {
+  const auto& config = filter_config();
+  std::string out_str;
+  TextFormat::PrintToString(config, &out_str);
```

Comment:
serialized computation is invoked for every request header as data path, might not added for now?

if WASM has some feature to load runtime feature (log level) and we can then put that computation there? same thing, i think logDebug is a function not a macro, and the DebugString is always invoked?

Reply:
I think this is another issue that should be resolved in the WASM level. I can just remove this for now.

Reply:
sounds good. might worth file a feature request.

/cc @PiotrSikora @jplevyak